### PR TITLE
Move upload storage from TestCafe to the hammerhead (closes #9)

### LINKED ISSUE #9: Implement upload storage
We should move upload storage from TestCafe to Hammerhead (in `src/upload`). Session should create it's storage instance with the provided `uploadsDir` and use it in the related service message handlers.

### PR DESCRIPTION
**Pull request by churkin**
\cc @inikulin @AlexanderMoskovkin @LavrovArtem @miherlosev 

superroma/testcafe-hammerhead#9

### CODE REVIEW

**Comment by inikulin** [comment]
BTW, we have related issue: #9. The title of the PR should contain `closes` or `fixes` keywords for this issue.

**Comment by churkin** [comment]
FPR

**Comment by inikulin** [comment]
Tests fail

**Comment by churkin** [comment]
@inikulin this is unstable test has already been fixed in the @miherlosev PR

**Comment by inikulin** [comment]
Ok, then squash commits and rebase, please

**Comment by churkin** [comment]
> @inikulin Why we need manually add trailing slashes here?

 https://github.com/churkin/testcafe-hammerhead/commit/59935527b5842567fb7fdc013b5925b05a2a9250 we do not need

**Comment by churkin** [comment]
@inikulin done

**Comment by churkin** [comment]
> In our last conversation we've decided what storage tests will be in upload-test.js or am I missing smth?

Can I remember in the different files

**Comment by inikulin** [comment]
@churkin Nope. In the same file and different sub-fixtures

**Comment by inikulin** [comment]
@churkin We have a massive test fails in IE9, consider filling an issue for this unstable tests?

**Comment by churkin** [comment]
@inikulin For the first time this test failing

**Comment by churkin** [comment]
FPR

**Comment by inikulin** [comment]
Squash commits pls

**Comment by churkin** [comment]
FPR

**Comment by inikulin** [comment]
\r- But in general just some minor style nits. And rebase pls.

File: test/server/upload-test.js [hunk 1]
```diff
@@ -249,6 +252,135 @@ describe('Upload', function () {
...
+            expect(fileInfo.info.lastModifiedDate).eql(fs.statSync(filePath).mtime);
+        }
+
+        it('Should store and get file', function (done) {
+            var storage = new UploadStorage(tmpDirObj.name);
+
+            var srcFilePath    = getSrcFilePath('file-to-upload.txt');
+            var storedFilePath = getStoredFilePath('file-to-upload.txt');
+
+            storage
```
Comment (inikulin) [inline test/server/upload-test.js]: Here and after: promise allows you avoid using such nested constructs (it's actually their puprose):
```js
storage
    .store(['file-to-upload.txt'], [filePath])
    .then(function (result) {
        expect(result.length).eql(1);
        assertCorrectErr(result[0], storedFilePath);

        return storage.get(['file-to-upload.txt']);
    })
    .then(function (result) {
        expect(result.length).eql(1);
        assertCorrectErr(result[0], storedFilePath);
        done();
    })
    .catch(done);
```

**Comment by inikulin** [comment]
\r-

**Comment by inikulin** [comment]
LGTM, nice job!

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by inikulin** [comment]
ping @LavrovArtem @miherlosev

**Comment by LavrovArtem** [comment]
lgtm

**Comment by miherlosev** [comment]
lgtm

**Pull request merged**